Adding chores api call with default userId

SalmaAnany opened the pull request.

SalmaAnany committed:
instailling MUI not working

SalmaAnany committed:
Port chours and MUI build tools
Need api calls for chores

# Conflicts:
#	src/App.tsx
#	src/Components/MissionPage/MissionView.tsx

SalmaAnany committed:
Chores Calling backend with default userId

SalmaAnany committed:
adding pnpm for build

Msambere commented on pnpm-lock.yaml:
I'm confused as to why there is a pnpm file. Since we aren't using the react-router framework I thought we don't need pnpm anymore.

Msambere commented on src/App.tsx:
```diff
@@ -1,44 +1,66 @@
+  });
+};
+
+export async function getChores(userId: number): Promise<ChoreResponse[]> {
+  try {
+    const response = await fetch(
+      `${VITE_APP_BACKEND_URL}/users/${userId}/chores`,
+    );
+    return await response.json();
+  } catch (error) {
+    console.error(error);
+    return [];
+  }
+}
```
Is there a reason you are using fetch instead of axios to make the API calls?

Msambere commented on src/Components/ChoresPage/ChoreCreate.tsx:
```diff
@@ -0,0 +1,119 @@
+import Autocomplete from "@mui/material/Autocomplete";
+import { Button } from "@mui/material";
+
+const previous_chore_tile = [
+  { previous_chore_tile: "Do Laundry" },
+  { previous_chore_tile: "Study Algorithm" },
+  { previous_chore_tile: "Go the gym" },
+];
+const recurrence = [
+  { recurrence: "Daily" },
+  { recurrence: "Weekly" },
+  { recurrence: "Monthly" },
+];
+
+const category = [
+  { category: "House Chores" },
+  { category: "Self Care" },
+  { category: "Self Study" },
+];
+
+const set_difficulty_level = [
+  { set_difficulty_level: "Easy" },
+  { set_difficulty_level: "Medium" },
+  { set_difficulty_level: "Hard" },
+];
```
We should remove all the hardcoded data now and just use the backend api.

Msambere commented on src/App.tsx:
```diff
@@ -1,44 +1,66 @@
+  const [chores, setChores] = useState<ChoreResponse[]>([]);
+  const [userId] = useState<number>(1);
 
   useEffect(() => {
-    getTestString().then((message ) => {
+    getChores(userId).then((chores) => setChores(chores));
+    getTestString().then((message) => {
       setMessage(message);
-  })
-}, []);
+    });
+  }, [userId, chores]);
```
I think we talked about this before, but  we should just make one API call that returns all the userdata instead of make separate API calls for chores, rewards, missions, etc.

Msambere commented on src/App.tsx:
```diff
@@ -1,44 +1,66 @@
 function App() {
   const [message, setMessage] = useState("");
-
+  const [chores, setChores] = useState<ChoreResponse[]>([]);
```
```suggestion
  const [message, setMessage] = useState<string>("");
  const [userData, setUserData] = useState([]) //Add typing
  const [userId, setUserID] = useState<number>(1);


  useEffect(() => {
    getTestString().then((message ) => {
      setMessage(message);
    });
    getUserDataAPICall(userId).then((data) =>{
      // default user is 1 for now
      setUserData(data);
      console.log(userData);
    })
}, []);

Msambere commented on src/App.tsx:
```diff
@@ -1,44 +1,66 @@
           {/*users see this page when they go to site, will be redirected if they are already logged in.*/}
           <Route path="/UserProfile" element={<UserProfileView />} />
-          <Route path="/Chores" element={<ChoresView />} />
+          <Route path="/Chores" element={<ChoresView chores={chores} />} />
```
```suggestion
          <Route path="/Chores" element={<ChoresView chores={userData.chores} />} />
```

After get all the userData from the backend, we can use donation to access the specific fields we need.

Msambere commented on src/Components/ChoresPage/ChoreView.tsx:
Can we remain this? It is too similar to ChoresView . If this component is making a list of chores, how about calling ChoreList.tsx?

Msambere commented on src/Components/LoginPage/LoginView.tsx:
Look like the changes are due to the linter

Msambere commented on src/Components/MissionPage/MissionView.tsx:
Looks like the changes are due to the linter

Msambere commented on src/Components/RewardsPage/RewardsView.tsx:
Look like the changes are due to the linter

Msambere commented on src/Components/SideBar/SideBar.tsx:
```diff
@@ -1,9 +1,29 @@
+      {menuItems.map((item) => (
+        <ListItem
+          key={item.label}
+          component={(props) => <Link {...props} to={item.route!} />}
```
Do we not need to use the `Link` or `NavLink` element from the react-router library here?

Msambere commented on src/Components/UserProfilePage/UserProfileView.tsx:
Look like the changes are due to the linter

Msambere commented on src/main.tsx:
```diff
@@ -1,10 +1,10 @@
+import { StrictMode } from "react";
+import { createRoot } from "react-dom/client";
+import "./index.css";
+import App from "~/App";
```
I'm curious why the import code is different now.

Msambere commented on src/types/ChoreResponse.ts:
Since I was writing the API call functions I also made interfaces for the request and response bodies. We need to talk about where/how to write and store them.

Msambere commented on src/types/MissionResponse.ts:
Since I was writing the API call functions I also made interfaces for the request and response bodies. We need to talk about where/how to write and store them.

Msambere commented on src/types/UserData.ts:
Since I was writing the API call functions I also made interfaces for the request and response bodies. We need to talk about where/how to write and store them.

Msambere commented on tsconfig.app.json:
Why do we have this file now?

Msambere commented on tsconfig.json:
Why did the file change so much?

Msambere commented on tsconfig.node.json:
Why is this file gone?

Msambere commented on vite.config.ts:
Can you explain the changes to me?

Msambere commented on src/Components/ChoresPage/ChoresView.tsx:
```diff
@@ -1,9 +1,33 @@
 
-export default ChoresView;
\ No newline at end of file
+export default function Chores({ chores }: ChoresProps) {
```
The function name should match the file name. Is there a reason you changed it from ChoresView to Chores?

Msambere commented on src/Components/ChoresPage/ChoreCreate.tsx:
Since this is a component not a function, I think it would be better to name it using a noun, not a verb. If this is the form for making a new chore, I suggest calling it ChoreForm.tsx

SalmaAnany committed:
Fixing callouts from amber
- Add Layout from old FE
- Rename some files
- move api call to apis

SalmaAnany committed:
Fixing callouts from amber
- Rename some files
- move api call to chores from
- add button to create a new chore
